docs/Features.md

```diff
@@ -0,0 +1,216 @@
+concatenated, in the order they are written. This can be very useful in defining global variables or
+listing imports closer to the parts where they are used.
+
+## Meta variables
+
+If you consider a macro invocation like a function call, then meta variables are like parameters.
+
+By default, to indicate that a macro includes a meta variable, the name of the variable must be part
+of the name of the macro, delimited by `@{` and `}`.
+
+Then that meta variable may be used within the macro by again using its name within the `@{` and `}`
+in the code.
+
+Finally, a macro with meta variables is invoked by replacing the name of the variable with its
+value in the invocation.
+
+An example:
+
+~~~md
+Here is our macro with meta variables:
+
+```rust - Say @{something} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say things to many people:
+
+```rust
+==> Say @{Hello} to @{Jim}.
+==> Say @{How are you} to @{Tom}.
+==> Say @{I am good!} to @{Angela}.
+```
+~~~
+
+Meta variables can have default values:
+
+~~~md
+Here is our macro with default meta variables:
+
+```rust - Say @{something:Hello} to @{someone}
+println!("Hey, @{someone}! I was told to tell you \"@{something}\"");
+```
+
+Now, to say the default "Hello" to many people:
+
+```rust
+==> Say @{} to @{Jim}.
+==> Say @{} to @{Tom}.
+==> Say @{} to @{Angela}.
+```
+~~~
+
+These features allows for more flexibility when writing macros, as well as possibly making the intent
+clearer.
+
+## Extracted comments
+
+By default, the comment extraction sequence is set to `//`, purely for familiarity. Any text after
+(and including) this sequence is extracted from the code block, and not rendered to the tangled
+source code. Note that, since the comments are removed completely when compiling, they do not have
+to use the actual line comment indicator from you programming language. In fact, it may be better to
+choose a sequence that is *not* the regular comment indicator so that you can still have comments in
+your tangled code output.
+
+Now that these comments are extracted, it is possible to handle them differently in the weaved
+documentation file. Though some formats do not support any special behaviour, and simply write these
+comments back into the code, some are able to provide special rendering. In particular, the standard
+Markdown and HTML styles are able to render extracted comments in `<aside>` tags, which can then be
+rendered nicely using CSS.
+
+See the HTML example for an example of one way to render the extracted comments.
+
+## Named entrypoints
+
+By default, the entrypoint of the program is always the unnamed code block. However, this limits the
+output of one input file to always be the same source code. It also means that you can't have a name
+on the entrypoint in the documentation, which could be useful.
+
+To get around this, an entrypoint name can be passed to Yarner on the command line. Then, instead
```

Comment:
Would it make sense to make the entry point name configurable via the TOML file so that I do not need to pass it on the command line for each invocation?

Reply:
Yes, this makes sense. Although it is probably rarely used. How was that with Some/None in TOML...

Reply:
`serde` and `toml` should handle `Option<String>` as the field type fine to support this optional item.